src/main/java/org/yorm/util/RowRecordConverter.java

```diff
@@ -1,90 +1,343 @@
 package org.yorm.util;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yorm.exception.YormException;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.math.BigDecimal;
-import java.sql.Date;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Time;
-import java.sql.Timestamp;
+import java.sql.*;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
-import org.yorm.exception.YormException;
+import java.util.Map;
 
 public class RowRecordConverter {
+    private static String test =
+            """
+                
+            """;
+    private static final Logger logger = LoggerFactory.getLogger(RowRecordConverter.class);
 
-    public void recordToRow(int paramIndex, PreparedStatement preparedStatement, String dbColumnName, Object value, DbType type) throws SQLException, YormException {
-        switch (type) {
-            //MySql does not have a truly boolean type, bool/boolean are a synonym of tinyint(1)
-            //Postgresql maps booleans to bits
-            case TINYINT, BIT, BOOLEAN -> preparedStatement.setBoolean(paramIndex, (boolean) value);
-            case SMALLINT, INTEGER -> {
-                if (value instanceof Integer) {
-                    preparedStatement.setInt(paramIndex, (int) value);
-                } else if (value instanceof Long) {
-                    preparedStatement.setLong(paramIndex, (long) value);
-                } else {
-                    throw new YormException("Incompatible value:" + value + " of class:" + value.getClass().getName() + " for column type:" + type);
+    private static final Map<Class<?>, Converter<?>> PASS_THROUGH_CONVERTERS = Map.ofEntries(
+            Map.entry(boolean.class, input -> input),
+            Map.entry(byte.class, input -> input),
+            Map.entry(char.class, input -> input),
+            Map.entry(short.class, input -> input),
+            Map.entry(int.class, input -> input),
+            Map.entry(long.class, input -> input),
+            Map.entry(float.class, input -> input),
+            Map.entry(double.class, input -> input),
+            Map.entry(Boolean.class, input -> input),
+            Map.entry(Byte.class, input -> (byte) input),
```

Comment:
Here we could use method reference:

Map.entry(Boolean.class, Boolean.class::cast),
            Map.entry(Byte.class, Byte.class::cast),
            Map.entry(Character.class, Character.class::cast),
            Map.entry(Short.class, Short.class::cast),
            Map.entry(Integer.class, Integer.class::cast),
            Map.entry(Long.class, Long.class::cast),
            Map.entry(Float.class, Float.class::cast),
            Map.entry(Double.class, Double.class::cast),
            Map.entry(String.class, String.class::cast)

Reply:
I was actually having LOTS of trouble with boxing and unboxing and casting back and forth between primitives and boxed types! Things get really messy at runtime, and this was the only thing I could get working! Type erasure can be a real pain in the butt! I was actually this close 🤏 to just generating classes on the fly to do the conversion.

I think Jackson does similar things, and it'll probably be good for performance! Maybe another TODO!

But the casting just won't work, unfortunately... :(